cmd/lncli/cmd_pay.go

```diff
@@ -330,9 +337,32 @@ func sendPayment(ctx *cli.Context) error {
 		if err != nil {
 			return err
 		}
+		args = args.Tail()
 		req.FinalCltvDelta = int32(delta)
 	}
 
+	var payAddr []byte
+	switch {
+	case ctx.IsSet("pay_addr"):
+		payAddr, err = hex.DecodeString(ctx.String("pay_addr"))
+	case args.Present():
+		payAddr, err = hex.DecodeString(args.First())
+	}
+
+	if err != nil {
+		return err
+	}
+	// payAddr may be not required if it's a legacy invoice.
+	if len(payAddr) == 0 {
+		return sendPaymentRequest(ctx, req)
+	}
+
+	if len(payAddr) != 32 {
```

Comment:
appears this could be simplified to `len(payAddr) != 0 && len(payAddr) != 32` like it now is in the rpc server. main benefit would be to avoid splitting the control flow and having multiple places where `sendPaymentRequest` is invoked

Reply:
don't recall why it's written so...now fixed!